tslearn/clustering.py

```diff
@@ -307,22 +306,41 @@ def fit(self, X, y=None, sample_weight=None):
         ----------
         X : array-like of shape=(n_ts, sz, d)
             Time series dataset.
+        y : Ignored
+            not used, present here for API consistency by convention.
         sample_weight : array-like of shape=(n_ts, ) or None (default: None)
             Weights to be given to time series in the learning process. By
             default, all time series weights are equal.
         """
 
+        X = check_array(X, allow_nd=True)
+        X = check_dims(X, X_fit=None)
+
+        if sample_weight is not None:
+            sample_weight = check_array(sample_weight, ensure_2d=False)
+
+        max_attempts = max(self.n_init, 10)
```

Comment:
It will at least perform 10 iterations :), but you are right, I will add this to the docs!

Reply:
After inspecting the code again, it seems like self.n_init will define the maximum number of successful iterations, while max_attempts represents the maximum total iterations (successful and unsuccessful ones).

```python3
while n_successful < self.n_init and n_attempts < max_attempts:
    try:
        if self.verbose and self.n_init > 1:
            print("Init %d" % (n_successful + 1))
        n_attempts += 1
        self._fit_one_init(X_, x_squared_norms, rs)
        if self.inertia_ < min_inertia:
            best_correct_centroids = self.cluster_centers_.copy()
            min_inertia = self.inertia_
            self.n_iter_ = self.iter_
        n_successful += 1
    except EmptyClusterError:
        if self.verbose:
            print("Resumed because of empty cluster")
```

Not sure what to do here...

Reply:
@rtavenar would there be another more robust way to perform initialization instead of just trying it a number of times and hoping it works once. Also, `max_attempts` now depends on the value of `n_init`, is this also something we want?

Reply:
I would leave it like that for the moment (as long as we don't have any better option). Ideally, we should maybe implement better initialization strategies, such as kmeans++ but it would not be straight-forward to implement with a dedicated kernel (though probably not too difficult).

And the fact that `max_attempts` now depends on the value of `n_init` is OK imho.

Reply:
This  is the only comment I did not address. Shall we just leave it as is for this PR?

Reply:
Yep!